src/pipeline.rs

```diff
@@ -51,7 +71,14 @@ pub fn build_graphs(
 
     let mut line = String::new();
     while buffered.read_line(&mut line).unwrap() > 0 {
-        let split: Vec<&str> = line.trim().split('\t').collect();
+        let split: Vec<SmallVec<[String; SMALL_VECTOR_SIZE]>> = match &config.file_type {
+            FileType::JSON => read_json_columns(&line, &config.columns),
+
+            FileType::TSV => {
+              let row = line.trim().split('\t');
+              row.map(|c| c.split(' ').map(|s| String::from(s)).collect()).collect()
```

Comment:
I think creating new strings every time will degrade performance.

Reply:
Agreed. This is one of the main reasons I don't think this code is ready as-is.